testing/src/main/java/io/grpc/testing/StreamRecorder.java

```diff
@@ -127,4 +142,18 @@ public boolean awaitCompletion(int timeout, TimeUnit unit) throws Exception {
   public ListenableFuture<T> firstValue() {
     return firstValue;
   }
+
+  /**
+   * Waits for a value to be received from the stream since the last
+   * call to awaitNext() or construction.  Useful for testing streaming
+   * call patterns.
+   */
+  public void awaitNext() {
```

Comment:
Tests hanging when they fail can be horrible to debug. Pass in a timeout (in the form `long, TimeUnit`). You then either need to return a boolean of whether next occurred in time (and assertTrue() at each usage) or throw a `TimeoutException`.